`Communication`: Show search result in answers indicator

FelberMartin opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check all tasks by putting an x in the [ ] (don't: [x ], [ x], do: [x]). Remove not applicable tasks and do not leave them unchecked -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable, and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).


#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.


### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
When searching for a message content, also posts showed up that include a search match in some of its answer posts. This was confusing to users because the base post that is shown to the user, does not match the search query and there is now easy way for the users to tell that the match is in an answer post.


### Description
<!-- Describe your changes in detail -->
- Added an info for search results, in case the search query also matches with the answer contents
- Fixed the loading animation for the search, to not show the loading status and the previous search results at once
- Only show the clear button in the search bar when the query is not empty


### Steps for Testing
<!-- Please describe in detail how reviewers can test your changes. Make sure to take all related features and views into account! Below is an example that you can refine. -->
Prerequisites:
- 1 Student

1. Log in to Artemis
2. Navigate to any chat
3. Create a new reply to any post
4. Search for that reply content using the global search
5. See that the base post of your reply shows up, and a small info indicating that the answer contains a match to the search query

### Testserver States
You can manage test servers using [Helios](https://helios.aet.cit.tum.de/). Check environment statuses in the [environment list](https://helios.aet.cit.tum.de/repo/69562331/environment/list). To deploy to a test server, go to the [CI/CD](https://helios.aet.cit.tum.de/repo/69562331/ci-cd) page, find your PR or branch, and trigger the deployment.

### Review Progress
<!-- Each PR should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2


### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. Remove the section if you did not change the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->
![image](https://github.com/user-attachments/assets/dc977534-2e05-4f67-b36f-b14d67da3031)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced search experience: The interface now displays clear feedback when no posts are found and highlights matching results in post answers.
	- Improved content filtering: Your search term is now actively used to refine and display relevant posts and hints.
	- Enhanced language support: Search result hints are available in multiple languages for a better user experience.
	- Conditional rendering for buttons based on search term presence.

- **Style**
	- Refreshed visual cues: New styling for search hints provides a consistent and clearer visual indicator.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

julian-wls reviewed (dismissed):
Really nice, tested on TS3 and works as expected.
Code lgtm!

SimonEntholzer commented on src/main/webapp/app/communication/post/post.component.ts:
```diff
@@ -85,6 +85,7 @@ export class PostComponent extends PostingDirective<Post> implements OnInit, OnC
     // if the post is previewed in the create/edit modal,
     // we need to pass the ref in order to close it when navigating to the previewed post via post title
     modalRef = input<NgbModalRef | undefined>(undefined);
+    searchQuery = input<string | undefined>(undefined);
```
Would it work here to use `input<string>('')` instead here, avoiding the undefined? Or would you need to adapt the `courseWideSearchConfig().searchTerm` then?

SimonEntholzer commented on src/main/webapp/app/communication/posting-reactions-bar/posting-reactions-bar.component.html:
```diff
@@ -42,6 +42,15 @@
+    @if (showSearchResultInAnswersHint()) {
+        <div class="search-result-in-answers-info">
+            <fa-icon class="fa-xs align-self-center" [icon]="faInfoCircle" />
+            <span class="emoji-count px-2">
+                {{ 'artemisApp.metis.overview.searchResultInAnswersHint' | artemisTranslate }}
+            </span>
```
We generally should use the jhi-translate instead of the translate pipe https://docs.artemis.cit.tum.de/dev/guidelines/client.html#localization

SimonEntholzer commented on src/main/webapp/app/communication/post/post.component.ts:
```diff
@@ -217,11 +219,26 @@ export class PostComponent extends PostingDirective<Post> implements OnInit, OnC
 
+    updateShowSearchResultInAnswersHint() {
+        const searchQuery = this.searchQuery()?.toLowerCase();
+        if (searchQuery === undefined || searchQuery === '') {
```
I guess here you could simply use `!searchQuery`, but I think that's up to personal preference

SimonEntholzer reviewed (changes requested):
Code looks good generally, I left a few remarks. You might also want to add a testcase, which covers the case when the search query isn't empty:
![image](https://github.com/user-attachments/assets/33b3b8b3-a4c0-4891-867e-b01870f69d5e)

FelberMartin commented:
> Code looks good generally, I left a few remarks. You might also want to add a testcase, which covers the case when the search query isn't empty

Thank you Simon for your valuable suggestions, these are especially helpful because I dont have much frontend dev experience yet. I addressed your code comments and added test cases as you suggested :D

ole-ve reviewed (dismissed):
reviewed the code, looks good 👍

cremertim reviewed (dismissed):
Reviewed code, lgtm. Great addition for the User experience 👍

ahmetsenturk reviewed (approved):
tested on TS2, works as described, cool feature 😊

PaRangger reviewed (approved):
Tested on TS4, very nice addition. 👍
